### ISSUE TITLE
Scorecard.score() silently uses truncated WoE values as bin indices when a per-variable metric is configured

### ISSUE THREAD

**Issue by zhongsheng-chen** [guillermo-navas-palencia/optbinning#412]
# Scorecard.score() silently uses truncated WoE values as bin indices when a per-variable metric is configured

### Description

With OptBinning 0.21.0, setting a per-variable `metric="woe"` in `BinningProcess.binning_transform_params` causes `Scorecard.score()` to silently look up the wrong bin.

The fitted variable correctly identifies the bin for `age=35` as index `1`. However, requesting `metric="indices"` from the fitted binning process returns `0`, which is the integer-truncated WoE value (`0.045587`). As a result, `Scorecard.score()` returns the points for bin 0 instead of bin 1.

### Environment

- OptBinning: 0.21.0
- Python: 3.12.4
- OS: Windows
- NumPy: 2.5.1
- pandas: 3.0.3
- scikit-learn: 1.9.0

### Minimal reproducible example

This example trains a single-feature scorecard entirely in memory using deterministic synthetic data. It does not require any model files, databases, or application services.

```python
"""Reproduce the WoE/bin-index conflict in OptBinning score calculation.

Usage:
  python examples/scorecard/reproduce_metric_conflict.py
"""

from importlib.metadata import version

import numpy as np
import pandas as pd
from optbinning import BinningProcess, Scorecard
from sklearn.linear_model import LogisticRegression


def main() -> None:
    """Compare the expected bin index, process output, and resulting scores."""
    frame = pd.DataFrame({"age": np.repeat([20, 35, 50], 100)})
    labels = np.concatenate([
        np.r_[np.zeros(100 - events), np.ones(events)]
        for events in (22, 32, 45)
    ]).astype(int)
    process = BinningProcess(
        variable_names=["age"],
        binning_fit_params={
            "age": {"user_splits": [30, 40], "user_splits_fixed": [True, True]},
        },
        binning_transform_params={
            "age": {
                "metric": "woe",
                "metric_missing": "empirical",
                "metric_special": "empirical",
            },
        },
        n_jobs=1,
    )
    model = Scorecard(
        binning_process=process,
        estimator=LogisticRegression(),
        scaling_method="pdo_odds",
        scaling_method_params={"pdo": 40, "odds": 45, "scorecard_points": 650},
    ).fit(frame, labels, metric_missing="empirical", metric_special="empirical")

    sample = pd.DataFrame({"age": [35]})
    fitted = model.binning_process_
    variable = fitted.get_binned_variable("age")
    options = {"metric_missing": "empirical", "metric_special": "empirical"}
    woe = float(variable.transform(sample["age"], metric="woe", **options)[0])
    expected_index = int(variable.transform(sample["age"], metric="indices", **options)[0])
    actual_index = int(fitted.transform(sample, metric="indices", **options)["age"].iloc[0])
    table = model.table(style="detailed")
    matched = table.loc[table["Bin id"] == expected_index].iloc[0]
    expected_score = float(matched["Points"]) + float(model.intercept_)
    actual_score = float(model.score(sample)[0])

    print(f"OptBinning: {version('optbinning')}")
    print("Input age: 35")
    print(f"Actual WoE: {woe:.6f}")
    print(f"Expected bin index: {expected_index}; bin: {matched['Bin']}")
    print(f"BinningProcess output when requesting indices: {actual_index}")
    print(f"WoE converted to integer: {int(woe)}")
    print(f"Expected score from the matched bin: {expected_score:.6f}")
    print(f"Actual Scorecard.score() result: {actual_score:.6f}")
    print("\nScorecard table:")
    print(table[["Bin id", "Bin", "WoE", "Points"]].to_string(index=False))
    if expected_index != actual_index:
        print("\nReproduced: requesting bin indices returns truncated WoE values, "
              "causing Scorecard.score() to look up the wrong bin.")
    else:
        print("\nThe index conflict was not reproduced in this environment. "
              "Please refer to the actual output above.")


if __name__ == "__main__":
    main()
```

### Actual output

```text
OptBinning: 0.21.0
Input age: 35
Actual WoE: 0.045587
Expected bin index: 1; bin: [30.00, 40.00)
BinningProcess output when requesting indices: 0
WoE converted to integer: 0
Expected score from the matched bin: 473.485803
Actual Scorecard.score() result: 500.705128

Scorecard table:
 Bin id            Bin       WoE     Points
      0  (-inf, 30.00)  0.557481 500.705128
      1 [30.00, 40.00)  0.045587 473.485803
      2   [40.00, inf) -0.507514 444.075375
      3        Special  0.000000 471.061788
      4        Missing  0.000000 471.061788
```

### Expected behavior

`Scorecard.score()` should use the bin actually matched by the input and return approximately `473.485803` for this example.

If explicit per-variable transformation settings are intentionally allowed to override `metric="indices"`, the scoring path should obtain genuine indices independently or reject the incompatible configuration rather than silently returning an incorrect score.
